packages/did-client/src/did-client.ts

```diff
@@ -1,90 +1,308 @@
 import Axios from 'axios'
-import { DIDDocument, Interfaces } from "@verida/did-document"
-import Wallet from "./wallet"
+// import { DIDDocument, Interfaces } from "@verida/did-document"
+import { DIDDocument } from 'did-resolver'
+const deepcopy = require('deepcopy')
 
-export default class DIDClient {
+import { default as VeridaWallet } from "./wallet"
+import { Wallet } from '@ethersproject/wallet'
 
-    private endpointUrl: string
+// @todo: Link this in package.json
+import { VdaDID, BulkDelegateParam, BulkAttributeParam, DelegateTypes } from '@verida/vda-did'
+import { getResolver, verificationMethodTypes, interpretIdentifier } from '@verida/vda-did-resolver'
+import { CallType, VeridaContract, VeridaMetaTransactionConfig, VeridaSelfTransactionConfig } from '@verida/web3'
 
-    private privateKey?: Uint8Array
-    private did?: string
-    private wallet?: Wallet
+import { Provider } from '@ethersproject/providers'
 
-    constructor(endpointUrl: string) {
-        // Strip any trailing slash
-        this.endpointUrl = endpointUrl.replace(/\/$/,'')
+import { Resolver, ServiceEndpoint, VerificationMethod } from 'did-resolver'
+// import { Signer } from 'ethers'
+import { Signer } from '@ethersproject/abstract-signer';
+
+import { DIDDocument as VeridaDocument } from '@verida/did-document'
+// import { VeridaMetaTransactionConfig } from '@verida/web3/build/src/config'
+
+import {isObject, removeCommonItems, getUpdateListFromDocument} from './helpers'
+
+// Part of VeridaSelfTransactionConfig
+interface VeridaSelfTransactionConfigPart  {
+    signer?: Signer
+    privateKey?: string
+}
+
+export interface DIDClientConfig {
+    // For signing doc
+    veridaPrivateKey: string
+
+    // vda-did resolver - only
+    identifier : string // DID
+    chainName? : string 
+    chainId? : string | number
+
+    // common to vda-did resolver & vda-did self transaction config
+    provider? : Provider
+    rpcUrl? : string
+    registry?: string
+    web3?: any
+
+    // vda-did
+    callType: CallType
+
+    // self or meta transaction configuration
+    web3Config: VeridaSelfTransactionConfigPart|VeridaMetaTransactionConfig
+}
+
+// Interfaces for saving DID Document
+interface VerificationMethodBase {
+    id: string
+    type: string
+    controller: string
+}
+
+interface Controller extends VerificationMethodBase {
+    blockchainAccountId: string
+}
+
+/**
+ * Interface to DIDClient instance
+ */
+export interface DIDClient {
+
+    authenticate(privateKey: string): void;
+
+    getDid(): string | undefined;
+
+    getPublicKey(): string;
+
+    saveDocument(document: DIDDocument): Promise<boolean>;
+
+    getDocument(): Promise<DIDDocument | null>
```

Comment:
Rename to "get()" to match the previous interface to avoid breaking other protocol code

Reply:
done